### DISCUSSION TITLE
`NotNullWhen` is not accessible in an analyzer project by default. Can I use it in my analyzer projects?

### DISCUSSION THREAD

**Discussion by MartyIX** [dotnet/roslyn#66711] [Q&A] [answered]
I want to use the `NotNullWhen` attribute in my analyzer code. An example is:

```cs
public static bool TryGetDocumentationComment(this MemberDeclarationSyntax member, [NotNullWhen(true)] out DocumentationCommentTriviaSyntax? comment)
```

However, my analyzer project targets `<TargetFramework>netstandard2.0</TargetFramework>` and so I think the `NotNullWhen` attribute is not available by default.

Can I reference a nuget package to add support for the nullability attributes (i.e. `NotNullWhen` in my case)?

Thank you!

**Comment by RikkiGibson** [comment, marked as answer]
There is probably a nuget package which adds these for netstandard2.0 projects, but you can also just declare an internal attribute type with the appropriate namespace, name, and shape, and the compiler will pick it up.

The compiler does [this](https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/InternalUtilities/NullableAttributes.cs) internally when building for netstandard.

**Reply by MartyIX** [reply]
Just to rephrase it: One just copies https://github.com/dotnet/roslyn/blob/main/src/Compilers/Core/Portable/InternalUtilities/NullableAttributes.cs to his/her analyzer project and the nullable attributes are supported.

Thank you!
